R/colby_constructors.R

```diff
@@ -871,11 +883,63 @@ split_rows_by_cutfun <- function(lyt, var,
 #' subsets of the data. These numbers will not, and currently cannot, reflect
 #' alternate column observation counts provided by the `alt_counts_df`,
 #' `col_counts` or `col_total` arguments to \code{\link{build_table}}}
+#' 
+#' @name spl_context
 NULL
 
-
-
-
+#' @title Additional parameters within analysis and content functions 
+#' (`afun/cfun`)
+#' 
+#' @description 
+#' It is possible to add specific parameters to `afun` and `cfun`, in [analyze]
+#' and [summarize_row_groups] respectively. These parameters grant access to 
+#' relevant information like the row split structure (see [spl_context]) and the
+#' predefined baseline (`.ref_group`).
+#' 
+#' @details
+#' We list and describe here all the parameters that can be added to a custom 
+#' analysis function:
+#' 
+#' \describe{
+#'   \item{.N_col}{column-wise N (column count) for the full column being
+#'     tabulated within}
+#'   \item{.N_total}{overall N (all observation count, defined as sum of column
+#'     counts) for the tabulation}
+#'   \item{.N_row}{row-wise N (row group count) for the group of observations
+#'     being analyzed (ie with no column-based subsetting)}
+#'   \item{.df_row}{data.frame for observations in the row group being analyzed
+#'     (ie with no column-based subsetting)}
+#'   \item{.var}{variable that is analyzed}
+#'   \item{.ref_group}{data.frame or vector of subset corresponding to the
+#'     `ref_group` column including subsetting defined by row-splitting.
+#'     Optional and only required/meaningful if a `ref_group` column has been
+#'     defined}
+#'   \item{.ref_full}{data.frame or vector of subset corresponding to the
+#'     `ref_group` column without subsetting defined by row-splitting. Optional
+#'     and only required/meaningful if a `ref_group` column has been defined}
+#'   \item{.in_ref_col}{boolean indicates if calculation is done for cells
+#'     within the reference column}
+#'   \item{.spl_context}{data.frame, each row gives information about a
+#'     previous/'ancestor' split state. See \code{\link{spl_context}}}
+#'   \item{.alt_df}{data.frame, i.e. the `alt_count_df` after 
+#'     row splitting. It can be used with `.all_col_exprs` and `.spl_context`
+#'     information to retrieve current faceting, but for `alt_count_df`.}
```

Comment:
It is producing an error that informs you that it is the split of `alt_counts_df` that failed. I will make it more explicit in docs

Reply:
I thought about this and put a warning but I think it is not necessary. The documentation needed to highlight this nonetheless

Reply:
The warning is for something different (i.e. if the filtering by rows and/or cols gives nothing). After discussing yesterday, I think this is not supposed to be a warning, if you filter out all the variables, it is fine to have empty tables (also because it is always used for calculating counts, for which it makes sense to have none).